docs/control-structures.rst

```diff
@@ -679,22 +681,43 @@ the changes in the caller will always be reverted.
     which consumed all gas available to the call.
     Exceptions that use ``require`` used to consume all gas until before the Metropolis release.
 
+.. _revert-statement:
+
 ``revert``
 ----------
 
-The ``revert`` function is another way to trigger exceptions from within other code blocks to flag an error and
-revert the current call. The function takes an optional string
-message containing details about the error that is passed back to the caller
-and it will create an ``Error(string)`` exception.
+A direct revert can be triggered using the ``revert`` statement and the ``revert`` function.
+
+The ``revert`` statement takes a custom error as direct argument without parentheses:
+
+    revert CustomError(arg1, arg2);
 
-The following example shows how to use an error string together with ``revert`` and the equivalent ``require``:
+For backards-compatibility reasons, there is also the revert function, which uses parentheses
+and accepts a string:
+
+    revert();
+    revert("description");
+
+The error data will be passed back to the caller and can be caught there.
+Using ``revert()`` causes a revert without any error data while ``revert("description")``
+will create an ``Error(string)`` error.
+
+Using a custom error instance will usually be much cheaper than a string description,
+because you can use the name of the error to describe it, which is encoded in only
+four bytes. A longer description can be supplied via NatSpec which does not incur
```

Comment:
instead of `incur`, i would prefer `increase` - its easier to understand.

Reply:
I'd vote for `incur`. Seems natural to me in this context - it's often used in combination with `cost`.